sql/pgwire/types.go

```diff
@@ -64,7 +65,7 @@ func typeForDatum(d driver.Datum) pgType {
 		return pgType{oid.T_bool, 1, formatText}
 
 	case *driver.Datum_IntVal:
-		return pgType{oid.T_int8, 8, formatBinary}
+		return pgType{oid.T_int8, 8, formatText}
```

Comment:
It was a cockroachism. I missed the cited part of the docs that requires text format when the client doesn't specifically request binary (and the go client driver accepts binary whether it requested it or not).